distributed/http/tests/test_core.py

```diff
@@ -53,14 +67,20 @@ async def test_prometheus_api_doc(c, s, a):
             "dask_worker_transfer_bandwidth_median_bytes",
         }
 
-    implemented = scheduler_metrics | worker_metrics | crick_metrics
-
-    documented = set()
-    root_dir = pathlib.Path(__file__).parent.parent.parent.parent
-    with open(root_dir / "docs" / "source" / "prometheus.rst") as fh:
-        for row in fh:
-            row = row.strip()
-            if row.startswith("dask_"):
-                documented.add(row)
+    # Optional GIL monitoring metrics, config set and gilknocker installed.
+    gil_metrics = {
+        "dask_scheduler_gil_contention",
+        "dask_scheduler_gil_contention_total",
+        "dask_worker_gil_contention",
+        "dask_worker_gil_contention_total",
+    }
+    try:
+        import gilknocker  # noqa: F401
+    except ImportError:
+        # Documented w/ notes about config and gilknocker requirements
+        for gm in gil_metrics:
+            documented.remove(gm)
+        gil_metrics = set()
```

Comment:
This detects a documented but not implemented metric:
```suggestion
    try:
        import gilknocker  # noqa: F401

        gil_metrics = set()  # Already in worker_metrics
    except ImportError:
        gil_metrics = {
            "dask_scheduler_gil_contention_total",
            "dask_worker_gil_contention_total",
        }
```

Reply:
Sorry, not sure I understand, this won't cover the non total metrics, and it compares documented to implemented just below; both of which are created independently of eachother.